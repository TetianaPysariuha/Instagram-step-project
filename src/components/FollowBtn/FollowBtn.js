import { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { loggedUserReceiveData, updateUser } from '../../store/users/actionCreators';
import styles from './FollowBtn.module.scss';

function FollowBtn({ currentUserId, loggedUserId }) {
  const dispatch = useDispatch();
  const users = useSelector((store) => store.users.users);
  const followBy = useSelector((store) => store.users.loggedUser.followBy) || [];

  useEffect(() => {
    if (loggedUserId) {
      dispatch(loggedUserReceiveData(loggedUserId));
    }
  }, [dispatch, users, loggedUserId]);

  if (loggedUserId === currentUserId) {
    return (<span className={styles.yourAccSpan}>це ваш аккаунт</span>);
  }

  const handleClickSubscribe = (userId) => {
    let newFollowBy;
    if (followBy.includes(userId)) {
      newFollowBy = followBy.filter((el) => el !== userId);
    } else {
      newFollowBy = [...followBy];
      newFollowBy.push(userId);
    }
    dispatch(updateUser({ id: loggedUserId, data: { followBy: newFollowBy } }));
    /* dispatch(loggedUserReceiveData(loggedUserId)); */
  };

  return (
    <button
      onClick={() => {
        handleClickSubscribe(currentUserId);
      }}
      className={styles.followBtn}
      type="button"
    >
      {followBy.includes(currentUserId) ? 'Відписатись' : 'Підписатися'}
    </button>
  );
}

FollowBtn.prototype = {
  currentUserId: PropTypes.string.isRequired,
  loggedUserId: PropTypes.string.isRequired,
};

FollowBtn.defaultProps = {
  followBy: [],
};

export default FollowBtn;
