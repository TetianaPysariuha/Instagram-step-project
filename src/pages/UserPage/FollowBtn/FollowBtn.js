import PropTypes from 'prop-types';
import styles from './FollowBtn.module.scss';

function FollowBtn({ currentUserId, loggedUserId }) {
  if (loggedUserId === currentUserId) {
    return (<span className={styles.yourAccSpan}>це ваш аккаунт</span>);
  }
  // else if(currentUserId != followBy.indexOf...) { return(<button onClick=(() => {}) type="button">Відписатись</button>)}
  return <button className={styles.followBtn} type="button">Стежити</button>;
}

FollowBtn.prototype = {
  currentUserId: PropTypes.string.isRequired,
  loggedUserId: PropTypes.string.isRequired,
};

export default FollowBtn;
