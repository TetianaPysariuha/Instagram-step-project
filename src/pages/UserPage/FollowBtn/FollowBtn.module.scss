.yourAccSpan {
    margin-left: 20px;
    border-radius: 4px;
    padding: 2px 20px;
    font-weight: 600;
    font-size: 14px;
    border: 1px solid rgb(219, 219, 219);
}

.followBtn {
    margin-left: 20px;
    background-color: rgb(0, 149, 246);
    border: 1px solid rgb(0, 149, 246);
    color: rgb(250, 250, 250);
    text-decoration: none;
    border-radius: 4px;
    padding: 2px 30px;
    font-weight: 600;
    font-size: 14px;
}