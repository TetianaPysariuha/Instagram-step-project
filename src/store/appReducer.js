import { combineReducers } from 'redux';
import modalReducer from './modal/reducer';

const appReducer = combineReducers({
  modal: modalReducer,
});

export default appReducer;
