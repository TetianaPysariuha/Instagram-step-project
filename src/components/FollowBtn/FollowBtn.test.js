import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store';
import FollowBtn from './FollowBtn';

function Component({ currentUserId, loggedUserId }) {
  return (
    <Provider store={store}>
      <FollowBtn currentUserId={currentUserId} loggedUserId={loggedUserId} />
    </Provider>
  );
}

describe('FollowBtn renders', () => {
  test('Should FollowBtn render with equal id', () => {
    const { asFragment } = render(<Component currentUserId="12345" loggedUserId="12345" />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should FollowBtn render with different id', () => {
    const { asFragment } = render(<Component currentUserId="12345" loggedUserId="54321" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
