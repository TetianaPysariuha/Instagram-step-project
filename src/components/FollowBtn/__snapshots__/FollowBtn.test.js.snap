// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowBtn renders Should FollowBtn render with different id 1`] = `
<DocumentFragment>
  <button
    class="followBtn"
    type="button"
  >
    Підписатися
  </button>
</DocumentFragment>
`;

exports[`FollowBtn renders Should FollowBtn render with equal id 1`] = `
<DocumentFragment>
  <span
    class="yourAccSpan"
  >
    це ваш аккаунт
  </span>
</DocumentFragment>
`;
